src/odemis/acq/test/orsay_milling_test.py

```diff
@@ -0,0 +1,282 @@
+elif TEST_NOHW == "1":
+    CONFIG_ORSAY["host"] = "192.168.56.101"  # IP address of the simulated Orsay Physics Control Server
+    TEST_NOHW = True
+else:
+    raise ValueError("Unknown value of environment variable TEST_NOHW=%s" % TEST_NOHW)
+
+
+class TestMilling(unittest.TestCase):
+    """
+    Test milling functions
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW is True:
+            raise unittest.SkipTest("No hardware available.")
+
+        # start the backend
+        cls.scanner = model.getComponent(role="ion-beam")
+        cls.scanner.horizontalFoV.value = 100e-6  # m
+        cls.host = cls.scanner.parent.host
+
+    def _get_milling_state(self):
+        """
+        Gets current status of the milling process
+        :params miller: HybridPatternCreator in Orsay API
+        :return: current milling state
+        """
+        server = Connection(self.host)
+        miller = server.datamodel.HybridPatternCreator
+
+        return miller.MillingActivationState.Actual
+
+    def test_milling_one_iteration(self):
+        """
+        Test milling the whole field of view for 20 seconds and one iteration
+        """
+        # mill the whole fov for 20 seconds
+        f = mill_rectangle(rect=[0, 0, 1, 1],
+                           scanner=self.scanner,
+                           iteration=1,
+                           duration=20,  # s
+                           probe_size=5e-6,  # m
+                           overlap=[0, 0])
+
+        time.sleep(3)
+        self.assertTrue(f.running())
+        milling_state = self._get_milling_state()
+        self.assertEqual(milling_state, '1')
+
+        # the milling process should be finished within about 30 seconds
+        f.result(timeout=30)
+        milling_state = self._get_milling_state()
+        self.assertEqual(milling_state, '0')
+
+        f.result()
+        self.assertTrue(f.done())
+
+    def test_milling_multiple_iterations(self):
+        """
+        Test milling half of the field of view for three iterations and 10 seconds each
+        """
+        # mill top-right half of the fov for two times 15 seconds
+        f = mill_rectangle(rect=[0.5, 0.5, 1, 1],
+                           scanner=self.scanner,
+                           iteration=2,
+                           duration=15,  # s
+                           probe_size=2e-7,  # m
+                           overlap=[0, 0])
+
+        time.sleep(4)
+        self.assertTrue(f.running())
+        milling_state = self._get_milling_state()
+        self.assertEqual(milling_state, '1')
+
+        # the milling process should be finished within about 40 seconds
+        f.result(timeout=40)
+        milling_state = self._get_milling_state()
+        self.assertEqual(milling_state, '0')
+
+        f.result()
+        self.assertTrue(f.done())
+
+    def test_milling_inverted_rect(self):
+        """
+        Test milling a rectangle with inverted values. X1>X2 and Y1>Y2 in this case.
```

Comment:
I think this test is overarching test for inverted vertical line and inverted horizontal line. Is that the reason we are are testing inverted lines?

Reply:
Ideally the X2 and Y2 values are greater than X1 and Y1 values respectively. This one is already an edge-case that does overarch the lines cases. So yes, that's the reason we are not testing inverted lines.